From 516ac701b3242d6ca3acc7d66dab167d8716a592 Mon Sep 17 00:00:00 2001
From: AnilKumar Ch <anilkumar@ti.com>
Date: Tue, 6 Mar 2012 13:02:19 +0530
Subject: [PATCH 0810/1092] ARM: OMAP: AM33XX: CAN: d_can: Add hwmod data for
 am33xx device

This patch adds the hwmod data needed by the driver. By adding
DCAN base addresses, interrupt numbers for two d_can instances.

Signed-off-by: AnilKumar Ch <anilkumar@ti.com>
---
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c | 67 +++++++++++++++++++++++++++---
 arch/arm/plat-omap/include/plat/am33xx.h   |  6 ---
 2 files changed, 62 insertions(+), 11 deletions(-)

diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index 4f00471..f264bdf 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -52,6 +52,8 @@ static struct omap_hwmod am33xx_tpcc_hwmod;
 static struct omap_hwmod am33xx_tptc0_hwmod;
 static struct omap_hwmod am33xx_tptc1_hwmod;
 static struct omap_hwmod am33xx_tptc2_hwmod;
+static struct omap_hwmod am33xx_dcan0_hwmod;
+static struct omap_hwmod am33xx_dcan1_hwmod;
 static struct omap_hwmod am33xx_gpio0_hwmod;
 static struct omap_hwmod am33xx_gpio1_hwmod;
 static struct omap_hwmod am33xx_gpio2_hwmod;
@@ -125,6 +127,42 @@ static struct omap_hwmod am33xx_l3slow_hwmod = {
 	.slaves_cnt	= ARRAY_SIZE(am33xx_l3_slow_slaves),
 };
 
+/* L4 PER -> DCAN0 */
+static struct omap_hwmod_addr_space am33xx_dcan0_addrs[] = {
+	{
+		.pa_start	= 0x481CC000,
+		.pa_end		= 0x481CC000 + SZ_4K - 1,
+		.flags		= ADDR_TYPE_RT
+	},
+	{ }
+};
+
+static struct omap_hwmod_ocp_if am33xx_l4_per__dcan0 = {
+	.master		= &am33xx_l4per_hwmod,
+	.slave		= &am33xx_dcan0_hwmod,
+	.clk		= "dcan0_ick",
+	.addr		= am33xx_dcan0_addrs,
+	.user		= OCP_USER_MPU | OCP_USER_SDMA,
+};
+
+/* L4 PER -> DCAN1 */
+static struct omap_hwmod_addr_space am33xx_dcan1_addrs[] = {
+	{
+		.pa_start	= 0x481D0000,
+		.pa_end		= 0x481D0000 + SZ_4K - 1,
+		.flags		= ADDR_TYPE_RT
+	},
+	{ }
+};
+
+static struct omap_hwmod_ocp_if am33xx_l4_per__dcan1 = {
+	.master		= &am33xx_l4per_hwmod,
+	.slave		= &am33xx_dcan1_hwmod,
+	.clk		= "dcan1_ick",
+	.addr		= am33xx_dcan1_addrs,
+	.user		= OCP_USER_MPU | OCP_USER_SDMA,
+};
+
 /* L4 PER -> GPIO2 */
 static struct omap_hwmod_addr_space am33xx_gpio1_addrs[] = {
 	{
@@ -181,6 +219,8 @@ static struct omap_hwmod_ocp_if am33xx_l4_per__gpio3 = {
 
 /* Master interfaces on the L4_PER interconnect */
 static struct omap_hwmod_ocp_if *am33xx_l4_per_masters[] = {
+	&am33xx_l4_per__dcan0,
+	&am33xx_l4_per__dcan1,
 	&am33xx_l4_per__gpio1,
 	&am33xx_l4_per__gpio2,
 	&am33xx_l4_per__gpio3,
@@ -464,17 +504,23 @@ static struct omap_hwmod am33xx_cpgmac0_hwmod = {
 
 /* 'dcan' class */
 static struct omap_hwmod_class am33xx_dcan_hwmod_class = {
-	.name		= "dcan",
+	.name		= "d_can",
+};
+
+/* dcan0 slave ports */
+static struct omap_hwmod_ocp_if *am33xx_dcan0_slaves[] = {
+	&am33xx_l4_per__dcan0,
 };
 
 /* dcan0 */
 static struct omap_hwmod_irq_info am33xx_dcan0_irqs[] = {
-	{ .irq = 52 },
+	{ .name = "d_can_ms", .irq = 52 },
+	{ .name = "d_can_mo", .irq = 53 },
 	{ .irq = -1 }
 };
 
 static struct omap_hwmod am33xx_dcan0_hwmod = {
-	.name		= "dcan0",
+	.name		= "d_can0",
 	.class		= &am33xx_dcan_hwmod_class,
 	.clkdm_name	= "l4ls_clkdm",
 	.mpu_irqs	= am33xx_dcan0_irqs,
@@ -485,15 +531,24 @@ static struct omap_hwmod am33xx_dcan0_hwmod = {
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
+	.slaves		= am33xx_dcan0_slaves,
+	.slaves_cnt	= ARRAY_SIZE(am33xx_dcan0_slaves),
+};
+
+/* dcan1 slave ports */
+static struct omap_hwmod_ocp_if *am33xx_dcan1_slaves[] = {
+	&am33xx_l4_per__dcan1,
 };
 
 /* dcan1 */
 static struct omap_hwmod_irq_info am33xx_dcan1_irqs[] = {
-	{ .irq = 55 },
+	{ .name = "d_can_ms", .irq = 55 },
+	{ .name = "d_can_mo", .irq = 56 },
 	{ .irq = -1 }
 };
+
 static struct omap_hwmod am33xx_dcan1_hwmod = {
-	.name		= "dcan1",
+	.name		= "d_can1",
 	.class		= &am33xx_dcan_hwmod_class,
 	.clkdm_name	= "l4ls_clkdm",
 	.mpu_irqs	= am33xx_dcan1_irqs,
@@ -504,6 +559,8 @@ static struct omap_hwmod am33xx_dcan1_hwmod = {
 			.modulemode	= MODULEMODE_SWCTRL,
 		},
 	},
+	.slaves		= am33xx_dcan1_slaves,
+	.slaves_cnt	= ARRAY_SIZE(am33xx_dcan1_slaves),
 };
 
 /* debugss */
diff --git a/arch/arm/plat-omap/include/plat/am33xx.h b/arch/arm/plat-omap/include/plat/am33xx.h
index 3a6ac9b..a16e72c 100644
--- a/arch/arm/plat-omap/include/plat/am33xx.h
+++ b/arch/arm/plat-omap/include/plat/am33xx.h
@@ -43,9 +43,6 @@
 #define AM33XX_TSC_BASE		0x44E0D000
 #define AM33XX_RTC_BASE		0x44E3E000
 
-#define AM33XX_D_CAN0_BASE	0x481CC000
-#define AM33XX_D_CAN1_BASE	0x481D0000
-
 #define AM33XX_ASP0_BASE	0x48038000
 #define AM33XX_ASP1_BASE	0x4803C000
 
@@ -62,9 +59,6 @@
 #define AM33XX_SPI0_BASE	0x48030000
 #define AM33XX_SPI1_BASE	0x481A0000
 
-#define AM33XX_DCAN0_BASE	0x481CC000
-#define AM33XX_DCAN1_BASE	0x481D0000
-
 #define AM33XX_USBSS_BASE	0x47400000
 #define AM33XX_USB0_BASE	0x47401000
 #define AM33XX_USB1_BASE	0x47401800
-- 
1.7.11.2

